tools/gen/impl_template

```diff
@@ -105,12 +112,15 @@ ${static}bool fromwire_${subtype.name}(${'const tal_t *ctx, ' if subtype.has_len
 	%endif
 	% endfor
 
-	return cursor != NULL;
+	return ${subtype.name};
```

Comment:
This seems strange: isn't this supposed to return a `bool`?

Reply:
there's two different 'types' of subtype method declarations: 
  - a 'varsize'd subtype that allocates the subtype itself and returns it. this belongs to that 'type' of subtype fromwire method. a type that has variable length fields is considered 'varsized'. This mimics how the `varlen_structs` in `generate-wire.py` originally work.
 - a 'normal' subtype (no internal allocations). the method signature includes a pointer to the already allocated struct for that subtype. it returns nothing. This mimics the way that other, manually defined subtypes work.

Ostensibly, I could have used the same format for both of them, but chose to copy the existing handwritten parsing format.